sockeye/training.py

```diff
@@ -747,11 +757,16 @@ def _update_metrics(self,
                               "time-elapsed": time.time() - self.state.start_tic}
         gpu_memory_usage = utils.get_gpu_memory_usage(self.model.context)
         checkpoint_metrics['used-gpu-memory'] = sum(v[0] for v in gpu_memory_usage.values())
+        checkpoint_metrics['converged'] = self.state.converged
+        checkpoint_metrics['diverged'] = self.state.diverged
 
+        # copy values from checkpoint_metrics, skipping NaNs and infinities that break numpy's histograms
         for name, value in metric_train.get_name_value():
-            checkpoint_metrics["%s-train" % name] = value
+            if np.isfinite(value):
+                checkpoint_metrics["%s-train" % name] = value
```

Comment:
how about adding a warning here and below?

Reply:
Added